ambry-store/src/main/java/com/github/ambry/store/DiskSpaceAllocator.java

```diff
@@ -101,6 +103,12 @@
         prepareDirectory(swapReserveDir);
         inventoryExistingReserveFiles();
         poolState = PoolState.INVENTORIED;
+      } else {
+        // Clean up reserve directory if previously created.
+        if (reserveDir != null) {
+          deleteDirectory(reserveDir);
```

Comment:
you can use `deleteFileOrDirectory()` in `Utils.java`

Reply:
changed to our `deleteFileOrDirectory `